src/main/java/com/xjeffrose/xio/http/SimpleProxyHandler.java

```diff
@@ -6,86 +6,81 @@
 import com.xjeffrose.xio.server.Route;
 import com.xjeffrose.xio.tracing.HttpTracingState;
 import io.netty.channel.ChannelHandlerContext;
-import io.netty.handler.codec.http.DefaultFullHttpResponse;
-import io.netty.handler.codec.http.HttpClientCodec;
-import io.netty.handler.codec.http.HttpContent;
-import io.netty.handler.codec.http.HttpRequest;
-import io.netty.handler.codec.http.HttpResponseStatus;
-import io.netty.handler.codec.http.HttpUtil;
-import io.netty.handler.codec.http.HttpVersion;
-import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.handler.codec.http.*;
 import io.netty.util.AttributeKey;
 import io.netty.util.ReferenceCountUtil;
-import java.util.Optional;
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
+import java.util.Optional;
 
 @Slf4j
-public class SimpleProxyRoute implements RouteProvider {
+public class SimpleProxyHandler implements RequestHandler {
 
   private static final AttributeKey<XioClient> key = AttributeKey.newInstance("xio_client");
+  @Getter
   private final Route route;
+  @Getter
   private final ProxyConfig config;
   private final XioClientBootstrap bootstrap;
   private XioClient client;
 
-  public SimpleProxyRoute(Route route, ProxyConfig config, XioClientBootstrap bootstrap) {
-    this.config = config;
+  public SimpleProxyHandler(Route route, ProxyConfig config, XioClientBootstrap bootstrap) {
     this.route = route;
+    this.config = config;
     this.bootstrap = bootstrap;
   }
 
-  public SimpleProxyRoute(Route route, ProxyConfig config) {
+  public SimpleProxyHandler(Route route, ProxyConfig config) {
     this(route, config, new XioClientBootstrap());
   }
 
   private void buildAndAttach(ChannelHandlerContext ctx) {
-
     client = bootstrap.clone(ctx.channel().eventLoop())
       .address(config.address)
       .ssl(config.needSSL)
-      .applicationProtocol(() -> new HttpClientCodec())
+      .applicationProtocol(HttpClientCodec::new)
       .handler(new RawBackendHandler(ctx))
       .build()
     ;
     ctx.channel().attr(key).set(client);
   }
 
   @Override
-  public RouteUpdateProvider handle(HttpRequest payload, ChannelHandlerContext ctx) {
+  public RequestUpdateHandler handle(HttpRequest payload, ChannelHandlerContext ctx) {
     ReferenceCountUtil.retain(payload);
     buildAndAttach(ctx);
     if (HttpUtil.is100ContinueExpected(payload)) {
       ctx.writeAndFlush(new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.CONTINUE));
     }
 
-    Optional<String> path = route.groups(payload.getUri())
+    Optional<String> path = route.groups(payload.uri())
       .entrySet()
       .stream()
       .filter(e -> e.getKey().equals("path"))
       .map(e -> e.getValue())
       .findFirst();
 
-    payload.setUri(path.map(config.urlPath::concat).orElse(config.urlPath));
+    if (!config.pathPassthru) {
```

Comment:
Are we expecting a service to respond to the full url 

`https://api.nordstrom.com/v1/backend/foo`

instead of 

`/foo`

?

My preference is to simplify things for the backend service as much as possible

Reply:
this setting is for nfe --> nlp.  in this case we want the entire original url to pass through to the nlp because it uses the prefix same as nfe does.  nlp then does the path replacement as usual.

Reply:
agreed on keep it simple for backend service.

Reply:
fixed this.  added default router and eliminated the null case.